scripts/src/javatests/org/oppia/android/scripts/regex/RegexPatternValidationCheckTest.kt

```diff
@@ -2343,7 +2345,35 @@ class RegexPatternValidationCheckTest {
       )
   }
 
-  // TODO(#5075): Add test for drawables file that checks color uses only component colors
+  @Test
+  fun testFileContent_xmlDrawables_includesNonColorComponentReferences_fileContentIsNotCorrect() {
+    val prohibitedContent =
+      """
+        android:color="@color/component_color_shared_primary_text_color"
+        android:color="@color/color_defs_shared_primary_text_color"
+        android:color="@color/color_palette_primary_text_color"
+        android:color="#003933"
+      """.trimIndent()
+    tempFolder.newFolder("testfiles", "app", "src", "main", "res", "drawable")
+    val stringFilePath = "app/src/main/res/drawable/test_layout.xml"
+    tempFolder.newFile("testfiles/$stringFilePath").writeText(prohibitedContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    // Verify that all patterns are properly detected & prohibited.
+    assertThat(exception).hasMessageThat().contains(REGEX_CHECK_FAILED_OUTPUT_INDICATOR)
+    assertThat(outContent.toString().trim())
```

Comment:
Is this output backwards? I'd expect only the hex code to be the failing line, or am I misunderstanding something?

Reply:
Anything besides the @color/component_color_* should fail.